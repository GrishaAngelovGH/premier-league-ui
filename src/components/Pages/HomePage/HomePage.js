import PageHeader from './PageHeader'

const HomePage = () => (
    <div className='row container-fluid no-gutters'>
        <div className='col-md-12 col-lg-7 mx-md-auto'>
            <div className='mt-4'>
                <PageHeader />
            </div>
        </div>
    </div>
)

export default HomePage