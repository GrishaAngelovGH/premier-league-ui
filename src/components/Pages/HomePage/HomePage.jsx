import PageHeader from './PageHeader'
import PageBody from './PageBody'
import PageFooter from './PageFooter'

const HomePage = () => (
	<div className="flex justify-center">
		<div className='w-[95%] lg:w-1/2 p-4'>
			<PageHeader />
			<PageBody />
			<PageFooter />
		</div>
	</div>
)

export default HomePage