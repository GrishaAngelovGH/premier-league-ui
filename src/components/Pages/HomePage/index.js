import HomePage, { matches } from './HomePage'
import PageBody from './PageBody'

export {
    matches,
    PageBody
}

export default HomePage