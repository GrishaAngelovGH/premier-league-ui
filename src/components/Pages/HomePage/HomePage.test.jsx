import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import AppContext from "context"

import HomePage from './HomePage'

test('should render HomePage component', () => {
	const matches = [
		[
			{
				id: '11',
				firstTeam: { image: 'crystalPalace', name: 'Crystal Palace', score: 0 },
				secondTeam: { image: 'newcastle', name: 'Newcastle', score: 2 },
				day: 'FT',
				time: 'Yesterday'
			},
			{
				id: '22',
				firstTeam: { image: 'brighton', name: 'Brighton', score: 1 },
				secondTeam: { image: 'liverpool', name: 'Liverpool', score: 3 },
				day: 'FT',
				time: 'Yesterday'
			},
		]
	]

	const view = render(
		<MemoryRouter>
			<AppContext.Provider value={{ matches }}>
				<HomePage />
			</AppContext.Provider>
		</MemoryRouter>
	)

	expect(view).toMatchSnapshot()
})