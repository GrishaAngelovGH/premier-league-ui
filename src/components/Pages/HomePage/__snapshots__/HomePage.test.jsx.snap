// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render HomePage component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center"
      >
        <div
          class="w-[95%] lg:w-1/2 p-4"
        >
          <div
            class="bg-purple-900 text-white h-[130px] rounded-t-md p-2"
          >
            <div
              class="flex items-center gap-2"
            >
              <img
                class="w-[50px] h-[50px]"
                src="/src/images/pl-logo.png"
              />
              <div
                class="text-xl"
              >
                <div>
                  Premier League
                </div>
                <div
                  class="opacity-[0.5]"
                >
                  Matchday 10 of 38
                </div>
              </div>
            </div>
            <div
              class="flex justify-evenly p-4 overflow-auto"
            >
              <a
                class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b border-white hover:border-b-0"
                data-discover="true"
                href="/content"
              >
                matches
              </a>
              <a
                class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b-0"
                data-discover="true"
                href="/content"
              >
                news
              </a>
              <a
                class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b-0"
                data-discover="true"
                href="/content"
              >
                standings
              </a>
              <a
                class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b-0"
                data-discover="true"
                href="/content"
              >
                stats
              </a>
              <a
                class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b-0"
                data-discover="true"
                href="/content"
              >
                players
              </a>
            </div>
          </div>
          <div>
            <div
              class="flex flex-col md:flex-row text-sm md:text-base"
            >
              <div
                class="md:w-1/2 border"
              >
                <div
                  class="flex justify-between cursor-pointer hover:bg-gray-200"
                >
                  <div
                    class="p-2 flex flex-col"
                  >
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/crystal-palace.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Crystal Palace
                      </div>
                      <div>
                        0
                      </div>
                    </div>
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/newcastle.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Newcastle
                      </div>
                      <div>
                        2
                      </div>
                    </div>
                  </div>
                  <div
                    class="flex-1"
                  >
                    <div
                      class="h-full flex flex-col justify-center items-center"
                    >
                      <div>
                        FT
                      </div>
                      <div>
                        Yesterday
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="md:w-1/2 border"
              >
                <div
                  class="flex justify-between cursor-pointer hover:bg-gray-200"
                >
                  <div
                    class="p-2 flex flex-col"
                  >
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/brighton.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Brighton
                      </div>
                      <div>
                        1
                      </div>
                    </div>
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/liverpool.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Liverpool
                      </div>
                      <div>
                        3
                      </div>
                    </div>
                  </div>
                  <div
                    class="flex-1"
                  >
                    <div
                      class="h-full flex flex-col justify-center items-center"
                    >
                      <div>
                        FT
                      </div>
                      <div>
                        Yesterday
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col md:flex-row text-sm md:text-base"
            >
              <div
                class="md:w-1/2 border"
              >
                <div
                  class="flex justify-between cursor-pointer hover:bg-gray-200"
                >
                  <div
                    class="p-2 flex flex-col"
                  >
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/man-city.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Man. City
                      </div>
                      <div>
                        2
                      </div>
                    </div>
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/burnley.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Burnley
                      </div>
                      <div>
                        1
                      </div>
                    </div>
                  </div>
                  <div
                    class="flex-1"
                  >
                    <div
                      class="h-full flex flex-col justify-center items-center"
                    >
                      <div>
                        FT
                      </div>
                      <div>
                        Yesterday
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="md:w-1/2 border"
              >
                <div
                  class="flex justify-between cursor-pointer hover:bg-gray-200"
                >
                  <div
                    class="p-2 flex flex-col"
                  >
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/everton.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Everton
                      </div>
                      <div />
                    </div>
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/leeds.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Leeds United
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="flex-1"
                  >
                    <div
                      class="h-full flex flex-col justify-center items-center"
                    >
                      <div>
                        Today
                      </div>
                      <div>
                        19:30
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col md:flex-row text-sm md:text-base"
            >
              <div
                class="md:w-1/2 border"
              >
                <div
                  class="flex justify-between cursor-pointer hover:bg-gray-200"
                >
                  <div
                    class="p-2 flex flex-col"
                  >
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/west-brom.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        West Brom
                      </div>
                      <div />
                    </div>
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/sheffield.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Sheffield United
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="flex-1"
                  >
                    <div
                      class="h-full flex flex-col justify-center items-center"
                    >
                      <div>
                        Today
                      </div>
                      <div>
                        22:00
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="md:w-1/2 border"
              >
                <div
                  class="flex justify-between cursor-pointer hover:bg-gray-200"
                >
                  <div
                    class="p-2 flex flex-col"
                  >
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/southampton.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Southampton
                      </div>
                      <div />
                    </div>
                    <div
                      class="flex justify-center items-center gap-2 m-1 w-[200px]"
                    >
                      <img
                        class="w-[30px] h-[30px]"
                        src="/src/images/man-utd.png"
                      />
                      <div
                        class="w-[120px]"
                      >
                        Man United
                      </div>
                      <div />
                    </div>
                  </div>
                  <div
                    class="flex-1"
                  >
                    <div
                      class="h-full flex flex-col justify-center items-center"
                    >
                      <div>
                        Tomorrow
                      </div>
                      <div>
                        16:00
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col justify-center items-center"
          >
            <div
              class="w-full text-center border p-2 text-gray-500"
            >
              <div>
                All times are in Eastern European Time
              </div>
            </div>
            <a
              class="mt-2 flex flex-col items-center text-center text-decoration-none"
              data-discover="true"
              href="/content"
            >
              <svg
                class="bg-purple-900 text-white w-[30px] h-[30px] rounded-full p-2"
                fill=" white"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z"
                  fill-rule="evenodd"
                />
              </svg>
              <div>
                See more
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center"
    >
      <div
        class="w-[95%] lg:w-1/2 p-4"
      >
        <div
          class="bg-purple-900 text-white h-[130px] rounded-t-md p-2"
        >
          <div
            class="flex items-center gap-2"
          >
            <img
              class="w-[50px] h-[50px]"
              src="/src/images/pl-logo.png"
            />
            <div
              class="text-xl"
            >
              <div>
                Premier League
              </div>
              <div
                class="opacity-[0.5]"
              >
                Matchday 10 of 38
              </div>
            </div>
          </div>
          <div
            class="flex justify-evenly p-4 overflow-auto"
          >
            <a
              class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b border-white hover:border-b-0"
              data-discover="true"
              href="/content"
            >
              matches
            </a>
            <a
              class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b-0"
              data-discover="true"
              href="/content"
            >
              news
            </a>
            <a
              class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b-0"
              data-discover="true"
              href="/content"
            >
              standings
            </a>
            <a
              class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b-0"
              data-discover="true"
              href="/content"
            >
              stats
            </a>
            <a
              class="text-white text-decoration-none uppercase p-1 hover:bg-purple-800 hover:rounded-md border-b-0"
              data-discover="true"
              href="/content"
            >
              players
            </a>
          </div>
        </div>
        <div>
          <div
            class="flex flex-col md:flex-row text-sm md:text-base"
          >
            <div
              class="md:w-1/2 border"
            >
              <div
                class="flex justify-between cursor-pointer hover:bg-gray-200"
              >
                <div
                  class="p-2 flex flex-col"
                >
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/crystal-palace.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Crystal Palace
                    </div>
                    <div>
                      0
                    </div>
                  </div>
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/newcastle.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Newcastle
                    </div>
                    <div>
                      2
                    </div>
                  </div>
                </div>
                <div
                  class="flex-1"
                >
                  <div
                    class="h-full flex flex-col justify-center items-center"
                  >
                    <div>
                      FT
                    </div>
                    <div>
                      Yesterday
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="md:w-1/2 border"
            >
              <div
                class="flex justify-between cursor-pointer hover:bg-gray-200"
              >
                <div
                  class="p-2 flex flex-col"
                >
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/brighton.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Brighton
                    </div>
                    <div>
                      1
                    </div>
                  </div>
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/liverpool.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Liverpool
                    </div>
                    <div>
                      3
                    </div>
                  </div>
                </div>
                <div
                  class="flex-1"
                >
                  <div
                    class="h-full flex flex-col justify-center items-center"
                  >
                    <div>
                      FT
                    </div>
                    <div>
                      Yesterday
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col md:flex-row text-sm md:text-base"
          >
            <div
              class="md:w-1/2 border"
            >
              <div
                class="flex justify-between cursor-pointer hover:bg-gray-200"
              >
                <div
                  class="p-2 flex flex-col"
                >
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/man-city.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Man. City
                    </div>
                    <div>
                      2
                    </div>
                  </div>
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/burnley.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Burnley
                    </div>
                    <div>
                      1
                    </div>
                  </div>
                </div>
                <div
                  class="flex-1"
                >
                  <div
                    class="h-full flex flex-col justify-center items-center"
                  >
                    <div>
                      FT
                    </div>
                    <div>
                      Yesterday
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="md:w-1/2 border"
            >
              <div
                class="flex justify-between cursor-pointer hover:bg-gray-200"
              >
                <div
                  class="p-2 flex flex-col"
                >
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/everton.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Everton
                    </div>
                    <div />
                  </div>
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/leeds.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Leeds United
                    </div>
                    <div />
                  </div>
                </div>
                <div
                  class="flex-1"
                >
                  <div
                    class="h-full flex flex-col justify-center items-center"
                  >
                    <div>
                      Today
                    </div>
                    <div>
                      19:30
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col md:flex-row text-sm md:text-base"
          >
            <div
              class="md:w-1/2 border"
            >
              <div
                class="flex justify-between cursor-pointer hover:bg-gray-200"
              >
                <div
                  class="p-2 flex flex-col"
                >
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/west-brom.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      West Brom
                    </div>
                    <div />
                  </div>
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/sheffield.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Sheffield United
                    </div>
                    <div />
                  </div>
                </div>
                <div
                  class="flex-1"
                >
                  <div
                    class="h-full flex flex-col justify-center items-center"
                  >
                    <div>
                      Today
                    </div>
                    <div>
                      22:00
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="md:w-1/2 border"
            >
              <div
                class="flex justify-between cursor-pointer hover:bg-gray-200"
              >
                <div
                  class="p-2 flex flex-col"
                >
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/southampton.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Southampton
                    </div>
                    <div />
                  </div>
                  <div
                    class="flex justify-center items-center gap-2 m-1 w-[200px]"
                  >
                    <img
                      class="w-[30px] h-[30px]"
                      src="/src/images/man-utd.png"
                    />
                    <div
                      class="w-[120px]"
                    >
                      Man United
                    </div>
                    <div />
                  </div>
                </div>
                <div
                  class="flex-1"
                >
                  <div
                    class="h-full flex flex-col justify-center items-center"
                  >
                    <div>
                      Tomorrow
                    </div>
                    <div>
                      16:00
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col justify-center items-center"
        >
          <div
            class="w-full text-center border p-2 text-gray-500"
          >
            <div>
              All times are in Eastern European Time
            </div>
          </div>
          <a
            class="mt-2 flex flex-col items-center text-center text-decoration-none"
            data-discover="true"
            href="/content"
          >
            <svg
              class="bg-purple-900 text-white w-[30px] h-[30px] rounded-full p-2"
              fill=" white"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z"
                fill-rule="evenodd"
              />
            </svg>
            <div>
              See more
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
